test/core/end2end/run.sh

```diff
@@ -1,4 +1,5 @@
 #!/bin/sh
```

Comment:
the name `run.sh` doesn't seem very descriptive, `run_configured_test.sh`?

Reply:
Fair point. This script could do more than execute configured tests, it's a generic `"$@"`. This general script replaces two near-identical scripts that had overly-specific names, which I think may have hindered their reuse. It could more appropriately be `run_arguments.sh` or `run_anything.sh`. I could probably replace it with a genrule that does the same, but that would be over-engineering.

For what it's worth, this is not unprecedented, see `run.sh` in xds_k8s_test_driver for example. I think its location is sufficient enough context to make the name not terribly evil.